setup.py

```diff
@@ -143,10 +151,173 @@ def _header_bad(self, path):
                 bad = target not in line
         return bad
 
-here = os.path.dirname(__file__)
-with open(os.path.join(here, 'README.rst'), 'r') as fh:
+
+# Dependency checks
+# =================
+
+# GEOS
+try:
+    geos_version = subprocess.check_output(['geos-config', '--version'])
+    geos_version = tuple(int(v) for v in geos_version.split(b'.'))
+    geos_includes = subprocess.check_output(['geos-config', '--includes'])
+    geos_clibs = subprocess.check_output(['geos-config', '--clibs'])
+except (OSError, ValueError, subprocess.CalledProcessError):
+    warnings.warn(
+        'Unable to determine GEOS version. Ensure you have %s or later '
+        'installed, or installation may fail.' % (
+            '.'.join(str(v) for v in GEOS_MIN_VERSION), ))
+
+    geos_includes = []
+    geos_library_dirs = []
+    if sys.platform.startswith('win'):
+        geos_libraries = ['geos']
+    else:
+        geos_libraries = ['geos_c']
+else:
+    if geos_version < GEOS_MIN_VERSION:
+        print('GEOS version %s is installed, but cartopy requires at least '
+              'version %s.' % ('.'.join(str(v) for v in geos_version),
+                               '.'.join(str(v) for v in GEOS_MIN_VERSION)),
+              file=sys.stderr)
+        exit(1)
+
+    if PY3:
+        geos_includes = geos_includes.decode()
+        geos_clibs = geos_clibs.decode()
+
+    geos_includes = geos_includes.split()
+    geos_libraries = []
+    geos_library_dirs = []
+    for entry in geos_clibs.split():
+        if entry.startswith('-L'):
+            geos_library_dirs.append(entry[2:])
+        elif entry.startswith('-l'):
+            geos_libraries.append(entry[2:])
+
+
+# Proj4
+def find_proj_version_by_program(conda=None):
+    proj = find_executable('proj')
+    if proj is None:
+        print(
+            'Proj4 %s must be installed.' % (
+                '.'.join(str(v) for v in PROJ_MIN_VERSION), ),
+            file=sys.stderr)
+        exit(1)
+
+    if conda is not None and conda not in proj:
+        print(
+            'Proj4 %s must be installed in Conda environment "%s".' % (
+                '.'.join(str(v) for v in PROJ_MIN_VERSION), conda),
+            file=sys.stderr)
+        exit(1)
+
+    try:
+        proj_version = subprocess.check_output([proj],
+                                               stderr=subprocess.STDOUT)
+        proj_version = proj_version.split()[1].split(b'.')
+        proj_version = tuple(int(v.strip(b',')) for v in proj_version)
+    except (OSError, IndexError, ValueError, subprocess.CalledProcessError):
+        warnings.warn(
+            'Unable to determine Proj4 version. Ensure you have %s or later '
+            'installed, or installation may fail.' % (
+                '.'.join(str(v) for v in PROJ_MIN_VERSION), ))
+        proj_version = (0, 0, 0)
+
+    return proj_version
+
+
+conda = os.getenv('CONDA_DEFAULT_ENV')
```

Comment:
I'm not entirely comfortable with having conda special cases in the setup.py. I'm trying to get my head around what not having proj in pkg-config actually means.

Instead of checking for conda when we can't find the pkg-config information, is it worth seeing if we can find the proj executable, and working from there? After all, proj version can be determined with:

```
$> proj
Rel. 4.8.0, 6 March 2012
```